### PR TITLE
[mqtt.generic] create the proper item type for channels with units

### PR DESCRIPTION
**Pull request by ccutrer**
Fixes https://community.openhab.org/t/openhab-4-3-zigbee2mqtt-color-temperature-in-mired-not-working/160911/8

### CODE REVIEW

**Comment by ccutrer** [comment]
Build will fail until https://github.com/openhab/openhab-core/pull/4507 is merged.

**Comment by lolodomo** [comment]
Waiting for the merge of the required change in core framework.

I assume the fix should be backported into 4.3 ?

**Comment by andrewfg** [comment]
> the fix should be backported into 4.3

Yes. I think so. Both this PR in addons and the corresponding UnitUtils PR in core.

File: bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/NumberValue.java [hunk 1]
```diff
@@ -56,14 +57,25 @@ public class NumberValue extends Value {
...
         this.max = max;
         this.step = step == null ? BigDecimal.ONE : step;
         this.unit = unit;
     }
 
+    private static String getItemType(@Nullable Unit<?> unit) {
+        if (unit == null) {
+            return CoreItemFactory.NUMBER;
+        }
+        String dimension = UnitUtils.getDimensionName(unit);
```
Comment (andrewfg) [inline bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/NumberValue.java:72]: You need to add your special Mirek case handler here. And only call getDimensionName() for other units.

**Review by andrewfg — CHANGES_REQUESTED** [review]
As discussed in OH core

**Comment by ccutrer** [comment]
This is no longer dependent on the core PR

**Comment by lolodomo** [comment]
Nack-ported to 4.3.x branch.

**Comment by andrewfg** [comment]
@ccutrer congrats :) sorry that it was like squeezing water out of a stone :(

**Pull request merged**